x-pack/plugins/security_solution/public/common/components/event_details/summary_view.tsx

```diff
@@ -112,83 +122,124 @@ const getDescription = ({
 const getSummary = ({
   data,
   browserFields,
-  timelineId,
+  timelineId: contextId,
   eventId,
+  isDisplayingThreatSummary,
+  isDisplayingThreatInfo,
 }: {
   data: TimelineEventsDetailsItem[];
-  browserFields: BrowserFields;
+  browserFields?: BrowserFields;
   timelineId: string;
   eventId: string;
+  isDisplayingThreatSummary: boolean;
+  isDisplayingThreatInfo: boolean;
 }) => {
-  return data != null
-    ? fields.reduce<Summary>((acc, item) => {
-        const field = data.find((d) => d.field === item.id);
-        if (!field) {
-          return acc;
-        }
-        const linkValueField =
-          item.linkField != null && data.find((d) => d.field === item.linkField);
-        const linkValue = getOr(null, 'originalValue.0', linkValueField);
-        const value = getOr(null, 'originalValue.0', field);
-        const category = field.category;
-        const fieldType = get(`${category}.fields.${field.field}.type`, browserFields) as string;
-        const description = {
-          contextId: timelineId,
-          eventId,
-          fieldName: item.id,
-          value,
-          fieldType: item.fieldType ?? fieldType,
-          linkValue: linkValue ?? undefined,
-        };
-
-        if (item.id === 'signal.threshold_result.terms') {
-          try {
-            const terms = getOr(null, 'originalValue', field);
-            const parsedValue = terms.map((term: string) => JSON.parse(term));
-            const thresholdTerms = (parsedValue ?? []).map(
-              (entry: { field: string; value: string }) => {
-                return {
-                  title: `${entry.field} [threshold]`,
-                  description: {
-                    ...description,
-                    value: entry.value,
-                  },
-                };
-              }
-            );
-            return [...acc, ...thresholdTerms];
-          } catch (err) {
-            return acc;
-          }
-        }
+  if (!data) return [];
+  if (isDisplayingThreatSummary || isDisplayingThreatInfo) {
+    return data.reduce<Summary>((acc, { category, field, originalValue }) => {
+      if (isDisplayingThreatInfo && field === 'threat.indicator' && originalValue) {
+        const threatData = Array.isArray(originalValue)
+          ? originalValue.length === 1
+            ? JSON.parse(originalValue[0])
+            : originalValue.map((threatDataItem) => JSON.parse(threatDataItem))
+          : JSON.parse(originalValue);
+        return getDataFromSourceHits(threatData).map((threatInfoItem) => ({
```

Comment:
I _think_ it should be okay to use this method on the client since it's only going to be run on a single alert, but we should verify this with the threat hunting team and get their thoughts/opinions before merging.

/cc @elastic/security-threat-hunting